lib/ansible/cli/doc.py

```diff
@@ -438,6 +463,10 @@ def init_parser(self):
         exclusive.add_argument("--metadata-dump", action="store_true", default=False, dest='dump',
                                help='**For internal use only** Dump json metadata for all entries, ignores other options.')
 
+        self.parser.add_argument("--no-fail-on-errors", action="store_true", default=False, dest='no_fail_on_errors',
+                                 help='**For internal use only** Only used for --metadata-dump. '
```

Comment:
This looks good to me. I don't think there's an easy way to add a suboption for one choice in a mutually exclusive group with argparse, but you could raise an AnsibleOptionsError in run() if this option is not provded with --metadata-dump.

Reply:
That's a great idea! I've implemented it in 5b0f970dc74777c8f43844d2312c75f644e0ae2d.